### PR TITLE
refactor: abstracting the usage of the patch program to its own exe

### PR DESCRIPTION
**Pull request by erickzanardo**
## Description

The `patch` program was being used directly in the `artifact_manager`. To keep it consistent with the other executables it was extracted to its own dart file under the executables folder.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [ ] 🧪 Tests

### CODE REVIEW

File: packages/shorebird_cli/lib/src/executables/patch.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    );
+    final diffArguments = [
+      releaseArtifactPath,
+      patchArtifactPath,
+      diffPath,
+    ];
+
+    final result = await process.run(diffExecutable, diffArguments);
+
+    if (result.exitCode != 0) {
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/executables/patch.dart]:
```suggestion
    if (result.exitCode != ExitCode.success.code) {
```

File: packages/shorebird_cli/lib/src/executables/patch.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  @override
+  String toString() => message;
+}
+
+/// A wrapper around the `patch` executable.
+///
+/// Used to create diffs between files.
+///
+/// thorw PatchFailedException if the patch fails.
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/executables/patch.dart]:
```suggestion
/// Throws [PatchFailedException] if the patch command exits with non-zero code.
```

File: packages/shorebird_cli/lib/src/executables/patch.dart [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import 'package:path/path.dart' as p;
+import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/cache.dart';
+import 'package:shorebird_cli/src/shorebird_process.dart';
+
+/// A reference to a [PatchProgram] instance.
+final patchProgramRef = create(PatchProgram.new);
+
+/// The [PatchProgram] instance available in the current zone.
+PatchProgram get patchProgram => read(patchProgramRef);
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/executables/patch.dart]: I guess just calling it `Patch` would probably cause a naming conflict with our model object. Maybe `PatchExecutable` instead? @felangel wdyt?

Comment (erickzanardo) [reply packages/shorebird_cli/lib/src/executables/patch.dart]: Renamed to `PatchExecutable`

File: packages/shorebird_cli/diff [hunk 4]
```diff
@@ -0,0 +1 @@
+diff
```
Comment (bryanoltman) [inline packages/shorebird_cli/diff]: Did you mean to commit this?

**Review by bryanoltman — APPROVED** [review]

File: packages/shorebird_cli/diff (hunk 4 above)
Comment (erickzanardo) [reply packages/shorebird_cli/diff]: definitelly not, weird, I was sure I had removed, thanks for catching it, gonna remove and merge it

**Pull request merged**